tests/umock_c_int/umock_c_int.c

```diff
@@ -2911,4 +2911,90 @@ TEST_FUNCTION(CallCannotFail_sets_cannot_fail_for_expected_call)
     ASSERT_ARE_EQUAL(char_ptr, "", umock_c_get_actual_calls());
 }
 
+/* CaptureArgumentValue_{arg_name}(arg_type* arg_value) */
+
+/* Tests_SRS_UMOCK_C_LIB_01_210: [ If arg_value is NULL, umock_c shall raise an error with the code UMOCK_C_NULL_ARGUMENT. ]*/
+TEST_FUNCTION(CaptureArgumentValue_with_NULL_arg_value_indicates_an_error)
+{
+    // arrange
+    EXPECTED_CALL(test_dependency_1_arg_no_return(IGNORED_NUM_ARG))
+        .CaptureArgumentValue_a(NULL);
+
+    // act
+    test_dependency_1_arg_no_return(43);
+
+    // assert
+    // gah, there is a TFS item apparently since forever to fix this issue: the extra ERROR notification should not happen
+    ASSERT_ARE_EQUAL(size_t, 2, test_on_umock_c_error_call_count);
+    ASSERT_ARE_EQUAL(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR, test_on_umock_c_error_calls[0].error_code);
+    ASSERT_ARE_EQUAL(UMOCK_C_ERROR_CODE, UMOCK_C_NULL_ARGUMENT, test_on_umock_c_error_calls[1].error_code);
+}
+
+/* Tests_SRS_UMOCK_C_LIB_01_209: [ The CaptureArgumentValue_{arg_name} shall copy the value of the argument at the time of the call to arg_value. ]*/
+TEST_FUNCTION(CaptureArgumentValue_captures_the_argument_value)
+{
+    // arrange
+    int captured_arg_value = 42;
+
+    STRICT_EXPECTED_CALL(test_dependency_1_arg_no_return(IGNORED_NUM_ARG))
+        .CaptureArgumentValue_a(&captured_arg_value);
+
+    // act
+    test_dependency_1_arg_no_return(43);
+
+    // assert
+    ASSERT_ARE_EQUAL(int, 43, captured_arg_value);
+    ASSERT_ARE_EQUAL(char_ptr, "", umock_c_get_expected_calls());
+    ASSERT_ARE_EQUAL(char_ptr, "", umock_c_get_actual_calls());
+}
+
+/* Tests_SRS_UMOCK_C_LIB_01_209: [ The CaptureArgumentValue_{arg_name} shall copy the value of the argument at the time of the call to arg_value. ]*/
+TEST_FUNCTION(CaptureArgumentValue_does_not_capture_when_matching_does_not_happen)
```

Comment:
>does_not_capture_when_matching_does_not_happen [](start = 35, length = 46)

So effectively only if it is ignored will meaningfully capture it.

Because if it not ignored, then there has to be a value... and if the values do not match... then the test is assumed to error out somewhere later?

But querying the captured/matched is not a supported (yet) scenario of umock_c. 

What would it take to capture it anyway? Just asking :)

Reply:
That would mean we'd have to do this:
for each actual call we would have to look through all the expected calls (after the next expected one) and figure out which call would match by function name. For that call we'd have to do the capture and then mark it as used (so that subsequent calls do not attempt to also use that call)

It gets pretty hairy. So, for now it will stay where it is ...

---
In reply to: [266210741](https://github.com/Azure/umock-c/pull/55#discussion_r266210741) [](ancestors = 266210741)